### ISSUE TITLE
Kafka 0.8 Test Fails

### ISSUE THREAD

**Issue by LeiSun-1101** [SOHU-Co/kafka-node#835]
### Environment
* Node version: 6.11.4
* Kafka-node version: latest
* Kafka version: 0.8.2.2
### Question
Now I try to improve something of KafkaClient module. However, when I run the KafkaClient test with KAFKA_VERSION=0.8, some tests for the KafkaClient always fails. 

The "versions" test has some mechanism to disable the versions test for Kafka 0.8, but in fact for my machine it does not work. The "connect" tests also don't work for Kafka 0.8. My question is that if the KafkaClient is only used for Kafka (0.9+). If yes, should someone or I fix the KafkaClient test to adapt that or just leave it?  
I also create a pull request. #836

**Comment by hyperlink** [comment]
The tests purely ran against 0.9 before we added other versions of kafka. I'll take a look at your PR thanks!

**Comment by TysonAndre** [comment]
Possibly related: I think the latest releases (E.g. 2.3.2) might have issues connecting to kafka 0.8.0. I'm not sure what config settings (if any) cause it. KafkaClient.KafkaClient (with a list of kafka hosts) didn't work, but KafkaNode.Client (with a list of zookeeper nodes) did work.

I'm having issues connecting to a kafka  (2.1.0 to the latest 2.3.2, including 2.2.x) with kafka 0.8.

In kafka-node 2.0.1, connections to kafka 0.8.0 succeeded without any error messages, when I used the following snippet:

```javascript
const KafkaNode = require('kafka-node');
const KafkaClient = KafkaNode.KafkaClient;
const client = new KafkaClient({kafkaHost: 'kafkaexamplehost01.example.com:9092,kafkaexamplehost02.example.com:9092,kafkaexamplehost03.example.com:9092,kafkaexamplehost04.example.com:9092', connectTimeout: 1000, requestTimeout: 2000});

client.on('error', function(error) {
  console.error('kafka error', error);
});
client.on('connect', function() {
  console.log('connected to kafka');
});
client.on('ready', function() {
  console.log('ready to start publishing to kafka');
});

console.log(client);
```

In kafka-node 2.1.0 (But not 2.2), I got a strange error:

```
/path/to/example-project/node_modules/async/dist/async.js:955                                         
        if (fn === null) throw new Error("Callback was already called.");                                              
                         ^                                 
                                                                                                                       
Error: Callback was already called.                                                                                    
    at /path/to/example-project/node_modules/async/dist/async.js:955:32                               
    at next (/path/to/example-project/node_modules/async/dist/async.js:3008:18)                       
    at connectToBroker (/path/to/example-project/node_modules/kafka-node/lib/kafkaClient.js:160:11)   
    at onError (/path/to/example-project/node_modules/kafka-node/lib/kafkaClient.js:200:5)            
    at Socket.<anonymous> (/path/to/example-project/node_modules/kafka-node/lib/kafkaClient.js:214:5) 
    at emitOne (events.js:101:20)                          
    at Socket.emit (events.js:188:7)                                                                                   
    at emitErrorNT (net.js:1281:8)                                                                                     
    at _combinedTickCallback (internal/process/next_tick.js:80:11)                                                     
    at process._tickCallback (internal/process/next_tick.js:104:9)    
```

And in kafka-node 2.2.x and 2.3.x there were strange ECONNRESET errors for that example

```
KafkaClient { 
..........
  _eventsCount: 3 }          
connected to kafka                                                                                                     
kafka error { Error: read ECONNRESET                                                                                   
    at exports._errnoException (util.js:1020:11)                                                                       
    at TCP.onread (net.js:568:26) code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' }     
```

If there's anything you need to know about the contents of kafka.properties I can probably provide that.

**Comment by hyperlink** [comment]
@TysonAndre I never meant for `KafkaClient` to be used with 0.8. It has always been designed to work with 0.9 and above. If it worked in 0.8 it was by coincidence.

**Comment by TysonAndre** [comment]
I see. This would be nice to mention in https://github.com/SOHU-Co/kafka-node/blob/master/README.md#kafkaclient

**Comment by hyperlink** [comment]
Well there's a note about connecting directly to brokers being available only for 0.9+ :)   https://github.com/SOHU-Co/kafka-node#features
